### PR TITLE
Fix prediction async issue and load model per resource class.

### LINKED ISSUE #1161: Benchmark scio-tensorflow
Currently `PredictDoFn` loads one model per instance with async inference.

We should benchmarks:
 * blocking inference
 * model per class

### PR DESCRIPTION
**Pull request by regadas**
Closes #1161 #1224

### CODE REVIEW

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala [hunk 1]
```diff
@@ -110,16 +105,28 @@ private[tensorflow] class SavedBundlePredictDoFn[T, V](uri: String,
...
-      .thenApply[V](new Function[TensorFlowModel, V] {
+      .thenApplyAsync[V](new Function[TensorFlowModel, V] {
         override def apply(model: TensorFlowModel): V = f(model.instance().session().runner())
       })
       .toCompletableFuture
 
-  override def createResource(): ModelLoader[TensorFlowModel] = Models.tensorFlow(uri, options)
+  @Teardown
+  def teardown(): Unit = {
+    log.info(s"Closing down predict DoFn $this")
```
Comment (nevillelyh) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala]: `"Tearing down"`?

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala [hunk 2]
```diff
@@ -128,19 +135,30 @@ private[tensorflow] class GraphPredictDoFn[T, V](uri: String,
...
       .get()
-      .thenApply[V](new Function[TensorFlowGraphModel, V] {
+      .thenApplyAsync[V](new Function[TensorFlowGraphModel, V] {
         override def apply(model: TensorFlowGraphModel): V = f(model.instance().runner())
       })
       .toCompletableFuture
+
+  @Teardown
+  def teardown(): Unit = {
+    log.info(s"Closing down predict DoFn $this")
```
Comment (nevillelyh) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala]: `"Tearing down"`?

File: scio-core/src/main/java/com/spotify/scio/transforms/JavaAsyncDoFn.java [hunk 3]
```diff
@@ -18,18 +18,18 @@
 package com.spotify.scio.transforms;
 
 import com.google.common.collect.Iterables;
-import org.apache.beam.sdk.transforms.DoFn;
-
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 import java.util.function.Function;
+import org.apache.beam.sdk.transforms.DoFn;
```
Comment (nevillelyh) [inline scio-core/src/main/java/com/spotify/scio/transforms/JavaAsyncDoFn.java]: There's no actual import change. IntelliJ auto formatting?

Comment (regadas) [reply scio-core/src/main/java/com/spotify/scio/transforms/JavaAsyncDoFn.java]: yeah it was

**Comment by ravwojdyla** [comment]
@regadas could you share the benchmark numbers, specifically:
 * this fix vs last version
 * load per class
 * and i would be curious to completely remove async and just have a ModelLoader per class and concurrency through DoFn

**Comment by regadas** [comment]
using 1 n-standard-16 instance the avg predict wall time:

1) fixed async: 46 min
2) ModelLoader per class and DoFn concurrency: 36 min

**Pull request merged**